### ISSUE TITLE
Termtype rr:IRI not taken into account when rr:Constant is used

### ISSUE THREAD

**Issue by VBP8501** [ontop/ontop#580]
### Description

Hello. The termType for a rr:constant is incorrectly set to string instead of IRI when the R2RML mapping file uses a rr:Constant. This problem is not there when using the rr:Template. 

### Steps to Reproduce

Using the rr:predicateObjectMap where the objectMap is using a rr:constant and the termType is an IRI
```
    rr:predicateObjectMap [
        rr:predicate era:navigability;
        rr:objectMap [ rr:constant "http://data.europa.eu/949/concepts/navigabilities/rinf/Both"; rr:termType rr:IRI ];
    ].
```

**Expected behavior:** [What you expect to see]

Output triple:
`<subject_uri> era:navigability <http://data.europa.eu/949/concepts/navigabilities/rinf/Both> .`

**Actual behavior:** [What actually happens]

Output triple:
`<subject_uri> era:navigability "http://data.europa.eu/949/concepts/navigabilities/rinf/Both" .`

**Reproduces how often:** [Does it happen every time?]

Always

### Versions

I have seen this behavior in both the CLI 4.2.2 and the docker setup (latest tag at the moment of creation of this issue)

### Additional Information

Changing the R2RML mapping to use the rr:template (without any placeholder for column name) resolves the issue.
```
    rr:predicateObjectMap [
        rr:predicate era:navigability;
        rr:objectMap [ rr:template "http://data.europa.eu/949/concepts/navigabilities/rinf/Both"; rr:termType rr:IRI ];
    ].
```

**Comment by kontchakov** [comment]
Hi Mathias

Section 7.1 of the R2RML specification (https://www.w3.org/TR/r2rml/#constant) says that "If the constant-valued term map is an object map, then its constant value must be an IRI or literal." So, if you specify rr:termType rr:IRI, then the value of rr:constant should be enclosed in < > to make it an IRI, otherwise, the value in quotation marks is treated as a literal of type xsd:string. Section 7.4 contains a further clarification: "Constant-valued term maps are not considered as having a term type, and specifying rr:termType on these term maps has no effect. The type of the generated RDF term is determined directly by the value of rr:constant: If it is an IRI, then an IRI will be generated; if it is a literal, a literal will be generated."
 
In contrast to that Section 7.3 says "The value of the rr:template property must be a valid string template." So, even if you specify rr:termType rr:IRI, the value of rr:template should be a string.

To sum up: the correct way of using rr:constant with rr:termType rr:IRI is to enclose the IRI in < > instead of "".

Best
Roman

**Comment by VBP8501** [comment]
Hello Roman,

You are absolutely right. I must have read the documentation 2 times before posting this issue and still missed the relevant sections. Or I misinterpreted them... Sorry for this issue, but many thanks for your time!

**Comment by kontchakov** [comment]
No worries, Mathias, it's not the most intuitive part of the specification, but the rationale I suppose is as follows - in general, <> are used for IRIs and "" for literals. In the case of rr:template, however, the string is NOT strictly speaking an IRI because it can contain placeholders, so it has to be enclosed in "" rather than in <> even if the instantiated end-result is an IRI. But this is precisely why rr:termType is required - to indicate whether in the end it should be an IRI, blank node or a literal.

**Comment by VBP8501** [comment]
That reasoning indeed sounds very logic. Thank you for this insight!
